front/src/reducers/osrdconf/selectors.ts

```diff
@@ -1,31 +1,55 @@
+import {
+  OsrdMultiConfState,
+  OsrdConfState,
+  OsrdStdcmConfState,
+  MODES,
+} from 'applications/operationalStudies/consts';
 import { RootState } from 'reducers';
 
-export const getName = (state: RootState) => state.osrdconf.name;
+function getSection(state: RootState): OsrdConfState | OsrdStdcmConfState {
+  type confSections = Omit<OsrdMultiConfState, 'mode'>;
+  type SectionKey = keyof confSections;
+
+  const sectionMap: Record<string, SectionKey> = {
+    [MODES.simulation]: 'simulationConf',
+    [MODES.stdcm]: 'stdcmConf',
+  };
+
+  const section: SectionKey = sectionMap[state.osrdconf.mode];
+  return state.osrdconf[section];
+}
+
+export const getConf = (state: RootState) => getSection(state);
+export const getName = (state: RootState) => getSection(state).name;
```

Comment:
```suggestion
export const selectConf = (state: RootState) => getSection(state);
export const selectName = (state: RootState) => getSection(state).name;
```
For all the selectors, it would be better to use the select prefix rather than get (too general). That's what is recommended by redux documentation: https://redux.js.org/style-guide/#name-selector-functions-as-selectthing

Reply:
Initialy nammed by @anisometropie . to be discussed with other devs ?